### PR TITLE
Increase unit test coverage of small modules in raiden/transfer/

### LINKED ISSUE #3910: Increase test coverage on raiden/transfer/secret_registry.py
`raiden/transfer/secret_registry.py` currently has 76% test coverage. Increase it to 90%.

Coverage report: https://codecov.io/gh/raiden-network/raiden/tree/master/raiden

### PR DESCRIPTION
**Pull request by jomuel**
Some additional unit tests to increase the coverage some small modules:

- `transfer.utils` to 92%
- `transfer.secret_registry` to 100%
- `transfer.identifiers` to 81% 

The still uncovered statements in `transfer.identifiers` will mostly be removed with the switch to dataclasses.

Closes #3910.

### CODE REVIEW

File: raiden/tests/unit/transfer/test_utils.py [hunk 1]
```diff
@@ -17,3 +27,73 @@
...
+    block_hash = factories.make_block_hash()
+
+    events = events_for_onchain_secretreveal(settled, secret, 10, block_hash)
+    assert not events, "Secret reveal event should not be generated for settled channel"
+
+    settle = factories.replace(settle, result=TransactionExecutionStatus.FAILURE)
+    unusable = factories.create(factories.NettingChannelStateProperties(settle_transaction=settle))
+
+    events = events_for_onchain_secretreveal(unusable, secret, 10, block_hash)
+    assert not events, "Secret reveal event should not be generated for unusable channel."
```
Comment (hackaugusto) [inline raiden/tests/unit/transfer/test_utils.py]: Humm, what is an unusable channel here? This doesn't seem right to me.

File: raiden/tests/unit/transfer/test_utils.py [hunk 2]
```diff
@@ -17,3 +27,73 @@
...
+        ),
+        "balance_proof.locksroot": encode_hex(locksroot),
+        "balance_proof.sender": to_checksum_address(sender),
+    }
+    storage = Mock()
+
+    def check(arg):
+        assert check_dict_nested_attrs(arg, expected)
+
+    storage.get_latest_state_change_by_data_field = Mock(side_effect=check)
```
Comment (hackaugusto) [inline raiden/tests/unit/transfer/test_utils.py]: This is testing the Mock, not our real code. Please either remove it or use a real db.

Comment (jomuel) [reply raiden/tests/unit/transfer/test_utils.py]: Fine, I'll rewrite the test using a real db then. There's no way to reach 90% module coverage without it.

File: raiden/tests/unit/transfer/test_utils.py (hunk 1 above)
Comment (jomuel) [reply raiden/tests/unit/transfer/test_utils.py]: A channel with status `CHANNEL_STATE_UNUSABLE`, i. e. one with a failed settle or close transaction.

Comment (hackaugusto) [reply raiden/tests/unit/transfer/test_utils.py]: Ohhh, it took me a while. Yeah, if the channel is settled it doesn't make sense to register the secret, the window is over. (For some reason I thought it was just a closed channel)

**Pull request merged**